ocaml/utils/compilation_unit.ml

```diff
@@ -408,7 +408,7 @@ end = struct
             })
           arguments
       in
-      of_full (Global (Global_module.Name.create head arguments))
+      of_full (Global (Global_module.Name.create_exn head arguments))
```

Comment:
Should this handle errors properly? Are we assuming that currently only the build system can generate the syntax, and it's their responsibility to do it correctly?

Reply:
This is the general constructor for `Compilation_unit.t`s, so any problem with the arguments is going to be an internal error rather than bad user input. User input gets turned into a `Global_module.Name.t` to begin with, then goes through `of_global_name` rather than this code.